File: cpp/Program/Elektronik.cpp
#include <iostream>
#include "Barang.cpp"
using namespace std;

class Elektronik : public Barang
{
private:
    string merk;
    int garansi; // bulan

public:
    Elektronik() {}

    Elektronik(int id, string nama, int harga, string merk, int garansi)
        : Barang(id, nama, harga)
    {
        this->merk = merk;
        this->garansi = garansi;
    }

    // setter
    void setMerk(string merk) { this->merk = merk; }
    void setGaransi(int garansi) { this->garansi = garansi; }

    // getter
    string getMerk() { return merk; }
    int getGaransi() { return garansi; }
};

File: java/Program/Elektronik.java
public class Elektronik extends Barang {
    private String merk;
    private int garansi; // bulan

    public Elektronik() {
    }

    public Elektronik(int id, String nama, int harga, String merk, int garansi) {
        super(id, nama, harga);
        this.merk = merk;
        this.garansi = garansi;
    }

    // setter
    public void setMerk(String merk) {
        this.merk = merk;
    }

    public void setGaransi(int garansi) {
        this.garansi = garansi;
    }

    // getter
    public String getMerk() {
        return merk;
    }

    public int getGaransi() {
        return garansi;
    }
}
